src/cpp/ext/otel/otel_client_call_tracer.h

```diff
@@ -92,20 +92,19 @@ class OpenTelemetryPlugin::ClientCallTracer
     void RecordAnnotation(absl::string_view /*annotation*/) override;
     void RecordAnnotation(const Annotation& /*annotation*/) override;
     std::shared_ptr<grpc_core::TcpTracerInterface> StartNewTcpTrace() override;
-    void AddOptionalLabels(OptionalLabelComponent component,
-                           std::shared_ptr<std::map<std::string, std::string>>
-                               optional_labels) override;
+    void AddOptionalLabel(OptionalLabelKey key,
+                          grpc_core::RefCountedStringValue value) override;
 
    private:
     const ClientCallTracer* parent_;
     const bool arena_allocated_;
     // Start time (for measuring latency).
     absl::Time start_time_;
     std::unique_ptr<LabelsIterable> injected_labels_;
-    // The indices of the array correspond to the OptionalLabelComponent enum.
-    std::array<std::shared_ptr<std::map<std::string, std::string>>,
-               static_cast<size_t>(OptionalLabelComponent::kSize)>
-        optional_labels_array_;
+    // Preferring std::map<> over std::vector<std::pair<>> to account for
+    // possible duplicates additions.
+    std::map<OptionalLabelKey, grpc_core::RefCountedStringValue>
```

Comment:
Agreed. I was trying to avoid conversions between the enum and size_t, but that led to use of a map over an std::array. The benefits of an std::array where we avoid per-call allocations and get contiguous memory access vastly outweigh the conversion downside.